flink-runtime/src/test/java/org/apache/flink/runtime/dispatcher/TestingJobManagerRunnerRegistry.java

```diff
@@ -118,6 +122,93 @@ public static Builder builder() {
     }
 
+    /**
+     * Creates {@code Builder} that is initialized to similate a {@link JobManagerRunnerRegistry}
+     * implementation based on a single-{@link JobManagerRunner}. The default test implementation
+     * follows the {@code JobManagerRunnerRegistry} contract. This {@code JobManagerRunner} instance
+     * is held by the {@code AtomicReference}.
+     *
+     * @param singleRunnerReference the reference to the internally held {@code JobManagerRunner}.
+     */
+    public static Builder builderFromSingleEntry(
+            AtomicReference<JobManagerRunner> singleRunnerReference) {
+        return builder()
+                .withRegisterConsumer(
+                        jobManagerRunner -> {
+                            Preconditions.checkState(singleRunnerReference.get() == null);
+                            singleRunnerReference.set(jobManagerRunner);
+                        })
+                .withIsRegisteredFunction(
+                        jobId ->
+                                accessReference(
+                                        singleRunnerReference,
+                                        jobManagerRunner ->
+                                                jobManagerRunner.getJobID().equals(jobId),
+                                        () -> false))
+                .withGetFunction(
+                        jobId ->
+                                accessReference(
+                                        singleRunnerReference,
+                                        Function.identity(),
+                                        () -> throwNoSuchElementException(jobId)))
+                .withGetJobManagerRunnersSupplier(
+                        () -> CollectionUtil.ofNullable(singleRunnerReference.get()))
+                .withSizeSupplier(() -> singleRunnerReference.get() == null ? 0 : 1)
+                .withGetRunningJobIdsSupplier(
+                        () ->
+                                accessReference(
+                                        singleRunnerReference,
+                                        jobManagerRunner ->
+                                                Collections.singleton(jobManagerRunner.getJobID()),
+                                        Collections::emptySet))
+                .withUnregisterFunction(
+                        jobId ->
+                                accessReference(
+                                        singleRunnerReference,
+                                        jobManagerRunner -> {
+                                            if (jobManagerRunner.getJobID().equals(jobId)) {
+                                                return singleRunnerReference.getAndSet(null);
+                                            }
+
+                                            return throwNoSuchElementException(jobId);
+                                        },
+                                        () -> throwNoSuchElementException(jobId)))
```

Comment:
nit: we can use standard java api (optionals) here instead, which might be bit more readable as others are already familiar with that

```suggestion
                .withIsRegisteredFunction(
                        jobId ->
                                Optional.ofNullable(singleRunnerReference.get())
                                        .map(JobManagerRunner::getJobID)
                                        .map(jobId::equals)
                                        .orElse(false))
                .withGetFunction(
                        jobId ->
                                Optional.ofNullable(singleRunnerReference.get())
                                        .orElseThrow(() -> newNoSuchElementException(jobId)))
                .withGetJobManagerRunnersSupplier(
                        () -> CollectionUtil.ofNullable(singleRunnerReference.get()))
                .withSizeSupplier(() -> singleRunnerReference.get() == null ? 0 : 1)
                .withGetRunningJobIdsSupplier(
                        () ->
                                Optional.ofNullable(singleRunnerReference.get())
                                        .map(JobManagerRunner::getJobID)
                                        .map(Collections::singleton)
                                        .orElseGet(Collections::emptySet))
                .withUnregisterFunction(
                        jobId ->
                                Optional.ofNullable(singleRunnerReference.get())
                                        .map(JobManagerRunner::getJobID)
                                        .filter(jobId::equals)
                                        .map(ignored -> singleRunnerReference.getAndSet(null))
                                        .orElseThrow(() -> newNoSuchElementException(jobId)))
```

Reply:
Good suggestion 👍